metadata.yaml

```diff
@@ -5,23 +5,26 @@ description: |
   MySQL is a widely used, open-source relational database management system (RDBMS).
 summary: |
   Kubernetes charm for MySQL
-series:
-  - kubernetes
-storage:
-  db:
-    type: filesystem
-    location: /var/lib/mysql
-deployment:
-  type: stateful
-provides:
-  database:
-    interface: mysql
+bases:
```

Comment:
You can drop the `bases` section from `metadata.yaml`. It's populated automatically into `manifest.yaml` when you build the charm now :)

Reply:
When I remove `bases` section from `metadata.yaml` I get:

```bash
$ juju deploy ./mysql.charm --resource mysql-image=ubuntu/mysql:latest
ERROR Post https://localhost:33797/model/983d4c59-4bc6-440f-80ec-aec6c63e0184/charms?revision=0&schema=local&series=focal: cannot upload charm: invalid charm archive: containers with metadata v1 not valid
```

Reply:
Bases should end up in charmcraft.yaml to tell it how to build the image, and `charmcraft build/pack` should render the final content into a manifest.yaml that ends up in the .charm file.

Reply:
It could be because of the extra parameters on your oci image resource - trying just specifying `type` and `description`

Also double check your Juju version (and the controller version?)

You can see in https://github.com/jnsgruk/hello-kubecon a valid `metadata.yaml` with no `bases` section :)

Reply:
@balbirthomas had the same and [got it to work](https://github.com/canonical/prometheus-operator/pull/23#discussion_r633397945) without bases.

Reply:
Yes just make sure you have updated your charmcraft. If you installed by snap and are on the right (stable) channel this should have happend. May want to check that your juju controller is also at least 2.9.0.

Reply:
Thanks, DONE.

```bash
$ juju version
2.9.1-ubuntu-amd64

$ charmcraft version
0.10.0+113.gd6d5635
```